libsolidity/formal/BMC.cpp

```diff
@@ -453,6 +453,17 @@ void BMC::endVisit(FunctionCall const& _funCall)
 	}
 
 	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
+	if (_funCall.isSuffixCall() && funType.parameterTypes().size() == 2)
+	{
+		m_errorReporter.warning(
+			7520_error,
+			_funCall.location(),
+			"Literal suffix functions with two arguments are not yet supported by SMTChecker. "
+			"The value of the literal will not be affected by the suffix, which may lead to incorrect results."
+		);
+		return;
+	}
```

Comment:
I didn't see a good place to create a function that does this and eliminates the code duplication, so I just left it for now.

Reply:
Both modified functions call `SMTEncoder::endVisit(FunctionDefinition)`, so you could put it there instead and I think the behavior should be the same.
Also it should use `m_unsupportedErrors` instead of `m_errorReporter`.

Reply:
Maybe creating a `checkAndWarnUnsupportedSuffixFunction` in `SMTEncoder`?

Reply:
I would not worry too much about the warning being duplicated. It's temporary anyway.

Not sure about `SMTEncoder::endVisit(FunctionCall)`, I'd have to look deeper into the code, but one thing I can suggest is that you should find the failing assertion and bypass it. Or you could calculate the arguments that's wrong but simple - e.g. set the first one the the value of the literal and the second one to zero.

Reply:
Actually, like I said in another comment, you can probably just do the fractional decomposition and get rid of the warning because that will make the suffixes work properly.

Reply:
Ok yea, then it would be probably easy enough to actually fully support it.

Reply:
I think I did that in https://github.com/ethereum/solidity/pull/14118/commits/2ea0c9dcbf13702255708741175ce26c661d867f, but I am not sure if that is properly implemented, since I just created the expressions "on the fly" and did not add them do the `context`.

https://github.com/ethereum/solidity/blob/2ea0c9dcbf13702255708741175ce26c661d867f/libsolidity/formal/SMTEncoder.cpp#L3177-L3185

Reply:
Well, seems that this is what `constantExpr()` does too. It returns the value from `literalValue()` directly. And here the situation is even simpler because the decomposed arguments are not even expressions in the AST. They're just some constant numbers we get based on the constant input we have.

It looks to me like it should work but of course it's best if @leonardoalt can confirm.